### PR TITLE
ipn/ipnlocal: move URL validation to LocalBackend

### PR DESCRIPTION
**Pull request by maisem**
Updates tailscale/corp#7948

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

**Comment by andrew-d** [comment]
I think test failures are real, since they use a `http://localhost:1` URL?

File: control/controlclient/direct.go [hunk 1]
```diff
@@ -1705,28 +1702,6 @@ func (c *Direct) ReportHealthChange(sys health.Subsystem, sysErr error) {
 	res.Body.Close()
 }
 
-// validPopBrowserURL reports whether urlStr is a valid value for a
-// control server to send in a MapResponse.PopUpBrowserURL field.
-func (c *Direct) validPopBrowserURL(urlStr string) bool {
-	if urlStr == "" {
-		// Common case.
-		return true
-	}
```
Comment (bradfitz) [inline control/controlclient/direct.go:1714]: this is now wrong in this context after the move

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -1726,8 +1726,34 @@ func (b *LocalBackend) popBrowserAuthNow() {
...
 }
 
+// validPopBrowserURL reports whether urlStr is a valid value for a
+// control server to send in a *URL field.
+// b.mu must *not* be held.
+func (b *LocalBackend) validPopBrowserURL(urlStr string) bool {
+	if urlStr == "" {
+		// Common case.
+		return true
+	}
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:1736]: this is now wrong in this context after the move

File: ipn/ipnlocal/local.go [hunk 3]
```diff
@@ -1726,8 +1726,34 @@ func (b *LocalBackend) popBrowserAuthNow() {
...
+		// If the control server is using plain HTTP (likely a dev server),
+		// then permit http://.
+		return strings.HasPrefix(serverURL, "http://")
+	}
+	return false
+}
+
 func (b *LocalBackend) tellClientToBrowseToURL(url string) {
-	if url != "" {
+	if url != "" && b.validPopBrowserURL(url) {
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:1756]: but you get saved by this check at least

**Pull request merged**